ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -73,44 +76,77 @@ class MainActivity : ComponentActivity() {
+        // scrolling).
+        // ----------
+        // State for the ScalingLazyColumn (watch list screen)
         val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+        // State for the Column's vertical scroll modifier (watch detail screen).
+        val watchDetailScrollState: ScrollState = rememberScrollState()
 
         val swipeDismissableNavController = rememberSwipeDismissableNavController()
 
-        // Allows user to disable the text before the time and hide the vignette.
+        // Allows user to disable the text before the time.
         var showProceedingTextBeforeTime by rememberSaveable { mutableStateOf(false) }
-        var vignetteVisible by rememberSaveable { mutableStateOf(true) }
+
+        // Allows user to show/hide the vignette on appropriate screens.
+        // IMPORTANT NOTE: Usually you want to show the vignette all the time on screens with
+        // scrolling content, a rolling side button, or a rotating bezel. This preference is just
+        // to visually demonstrate the vignette for the developer to see it on and off.
+        var vignetteVisiblePreference by rememberSaveable { mutableStateOf(false) }
+
+        // Observes the current back stack entry to show the vignette for the appropriate screens by
+        // checking the destination + route.
+        // TODO: Replace with .currentBackStackEntryAsState() once added to Compose for Wear OS
+        // Navigation Library: https://issuetracker.google.com/issues/212739653
+        val currentBackStackEntry by swipeDismissableNavController
+            .currentBackStackEntryFlow.collectAsState(null)
+
+        val currentRoute = currentBackStackEntry?.destination?.route
 
         Scaffold(
             // Scaffold places time at top of screen to follow Material Design guidelines.
             timeText = {
                 CustomTimeText(
-                    visible = !scalingLazyListState.isScrollInProgress,
+                    visible = !scalingLazyListState.isScrollInProgress &&
+                        !watchDetailScrollState.isScrollInProgress,
                     showLeadingText = showProceedingTextBeforeTime,
                     leadingText = stringResource(R.string.leading_time_text)
                 )
             },
             vignette = {
-                CustomVignette(
-                    visible = vignetteVisible,
-                    vignettePosition = VignettePosition.TopAndBottom
-                )
+                // Only show vignette for screens with scrollable content.
+                if (Screen.WatchList.route in currentRoute) {
+                    CustomVignette(
+                        visible = vignetteVisiblePreference,
+                        vignettePosition = VignettePosition.TopAndBottom
+                    )
+                } else if (Screen.WatchDetail.route in currentRoute) {
+                    CustomVignette(
+                        visible = true,
```

Comment:
```suggestion
                        visible = vignetteVisiblePreference,
```

Did you want this to also use `vignetteVisiblePreference`?

Reply:
Not in this case, I always want the vignette to appear here. 

The only reason I let the user toggle it off/on in the list page was so they could see the visual difference... Usually we don't want them turning it off/on at all in an app... I wonder if I should just remove that feature to avoid confusion.

WDYT?

Reply:
I could go either way, but yeah I think removing anyway to hide it on a scrolling screen makes sense.

Reply:
My understanding though in general is we shouldn't remove the vignette on scroll. @JohnNichol is that correct?

Reply:
Correct - My understanding of the UX spec is that Vignette is always expected to be in place for scrolling content.